nifty-examples/src/test/java/com/facebook/nifty/server/TestNiftySSLServer.java

```diff
@@ -0,0 +1,149 @@
+import java.net.InetSocketAddress;
+import java.util.Arrays;
+import java.util.List;
+
+public class TestNiftySSLServer
+{
+    private static final Logger log = Logger.get(TestNiftySSLServer.class);
+    private NettyServerTransport server;
+    private int port;
+
+    @BeforeMethod(alwaysRun = true)
+    public void setup()
+    {
+        server = null;
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void teardown()
+            throws InterruptedException
+    {
+        if (server != null) {
+            server.stop();
+        }
+    }
+    private void startServer()
+    {
+        startServer(getThriftServerDefBuilder());
+    }
+
+    private void startServer(final ThriftServerDefBuilder thriftServerDefBuilder)
+    {
+        server = new NettyServerTransport(thriftServerDefBuilder.build(),
+                                          NettyServerConfig.newBuilder().build(),
+                                          new DefaultChannelGroup());
+        server.start();
+        port = ((InetSocketAddress)server.getServerChannel().getLocalAddress()).getPort();
+    }
+
+    private ThriftServerDefBuilder getThriftServerDefBuilder()
+    {
+        return new ThriftServerDefBuilder()
+                .listen(8080)
+                .withSSLConfiguration(
+                        new SSLServerConfiguration.Builder()
+                                .certFile("nifty-examples/certs/rsa.crt")
+                                .keyFile("nifty-examples/certs/rsa.key")
+                                .build())
+                .withProcessor(new scribe.Processor<>(new scribe.Iface() {
+                    @Override
+                    public ResultCode Log(List<LogEntry> messages)
+                            throws TException {
+                        RequestContext context = RequestContexts.getCurrentContext();
+
+                        for (LogEntry message : messages) {
+                            log.info("[Client: %s] %s: %s",
+                                    context.getConnectionContext().getRemoteAddress(),
+                                    message.getCategory(),
+                                    message.getMessage());
+                        }
+                        return ResultCode.OK;
+                    }
+                }));
+    }
+
+    private scribe.Client makeNiftyClient()
+            throws TTransportException, InterruptedException
+    {
+        NettyClientConfig config =
+                NettyClientConfig.newBuilder()
+                        .setSSLClientConfiguration(
+                                new SSLClientConfiguration.Builder()
+                                        .caFile("nifty-examples/certs/rsa.crt")
+                                        .build()).build();
+        InetSocketAddress address = new InetSocketAddress("localhost", port);
+        TTransport transport = new NiftyClient(config)
+                .connectSync(scribe.Client.class, new FramedClientConnector(address));
```

Comment:
Add a test that uses `.connectAsync`? This is the main interface, `.connectSync` is kinda deprecated